pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -295,6 +321,37 @@ func (p *staticPolicy) updateCPUsToReuse(pod *v1.Pod, container *v1.Container, c
 
 func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Container) (rerr error) {
 	numCPUs := p.guaranteedCPUs(pod, container)
+	// During a pod resize, handle corner cases
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		if v1qos.GetPodQOS(pod) != v1.PodQOSGuaranteed {
+			return nil
+		}
+		cpuQuantity := container.Resources.Requests[v1.ResourceCPU]
+		if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+			allocatedCPUQuantity := cs.AllocatedResources[v1.ResourceCPU]
+			if allocatedCPUQuantity.Value()*1000 == allocatedCPUQuantity.MilliValue() {
+				// container belongs in exclusive pool
+				if cpuQuantity.Value()*1000 != cpuQuantity.MilliValue() {
+					// container move to shared pool not allowed
+					return InPlacePodVerticalScalingError{
+						RequestedCPUs:    cpuQuantity.String(),
+						AllocatedCPUs:    allocatedCPUQuantity.String(),
+						Shared2Exclusive: false,
+					}
+				}
+			} else {
+				// container belongs in shared pool
+				if cpuQuantity.Value()*1000 == cpuQuantity.MilliValue() {
+					// container move to exclusive pool not allowed
+					return InPlacePodVerticalScalingError{
+						RequestedCPUs:    cpuQuantity.String(),
+						AllocatedCPUs:    allocatedCPUQuantity.String(),
+						Shared2Exclusive: true,
+					}
+				}
+			}
+		}
+	}
```

Comment:
I'd move this code into a helper, similar in spirit to `guaranteedCPUs`. Something like `func (p *staticPolicy) validateInPlacePodVerticalScaling(pod *v1.Pod, container *v1.Container) error`.
A different/better name is welcome.

Reply:
Done in latest patch

Reply:
@ffromani please let me know if we can resolve this